ROOT does not compile due to linking error with lssl

Error:
```
[ 80%] Linking CXX shared library ../../lib/libRMySQL.so
ld: library not found for -lssl
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/libRMySQL.6.14.04.so] Error 1
```

Environment:
- OS: macOS 10.14
- clang: clang-1000.11.45.2
- `brew --prefix openssl`: /usr/local/opt/openssl

Any ideas?

--- a/root.sh
+++ b/root.sh
@@ -42,6 +42,7 @@ COMPILER_LD=c++
 case $ARCHITECTURE in
   osx*)
     ENABLE_COCOA=1
+    DISABLE_MYSQL=1
     COMPILER_CC=clang
     COMPILER_CXX=clang++
     COMPILER_LD=clang
@@ -143,6 +144,7 @@ else
         -Dgviz=OFF                                                                       \
         -Dbuiltin_davix=OFF                                                              \
         -Ddavix=OFF                                                                      \
+        ${DISABLE_MYSQL:+-Dmysql=OFF}                                                    \
         -DCMAKE_PREFIX_PATH="$FREETYPE_ROOT;$SYS_OPENSSL_ROOT;$GSL_ROOT;$ALIEN_RUNTIME_ROOT;$PYTHON_ROOT;$PYTHON_MODULES_ROOT;$LIBPNG_ROOT;$LZMA_ROOT"
   FEATURES="builtin_pcre mathmore xml ssl opengl minuit2 http fortran
             pythia6 roofit soversion vdt ${CXX11:+cxx11} ${CXX14:+cxx14} ${CXX17:+cxx17}

